### PR TITLE
Adding pagerduty target

### PR DESCRIPTION
**Pull request by nevins-b**
This adds the ability to create pagerduty incidents from alerts

### CODE REVIEW

File: phplib/Target/PagerDuty.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+namespace FOO;
+
+/**
+ * PagerDuty Target Class
+ * Creates an event for an Alert.
+ */
+class PagerDuty_Target extends Target {
+    public static $TYPE = 'pagerduty';
+
+    public static $DESC = 'Create a incident for this alert and assign to <assignee>.';
```
Comment (kiwiz) [inline phplib/Target/PagerDuty.php]: Nitpicky: "an incident"

File: phplib/Target/PagerDuty.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * Create a PagerDuty Event.
+     * @param mixed[] $event_data Event data.
+     * @return string|null The incident key or null.
+     */
+    public static function createEvent($event_data) {
+        $curl = new \Curl\Curl;
+        $curl->setHeader('Content-type', 'application/json');
+        $ret = $curl->post(
+            'https://events.pagerduty.com/generic/2010-04-15/create_event.json',
```
Comment (kiwiz) [inline phplib/Target/PagerDuty.php]: Is that... the api version?

File: phplib/Target/PagerDuty.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            json_encode($event_data)
+        );
+
+        if($curl->httpStatusCode < 200 || $curl->httpStatusCode >= 300) {
```
Comment (kiwiz) [inline phplib/Target/PagerDuty.php]: Does the API return non 200 responses on success? If not, you should just check for 200 here.

File: phplib/Target/PagerDuty.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            return null;
```
Comment (kiwiz) [inline phplib/Target/PagerDuty.php]: I've updated the Jira Target to give better error info when a request fails: https://github.com/etsy/411/blob/master/phplib/Target/Jira.php#L83

**Comment by nevins-b** [comment]
So I'm not sure what to do for the  isAccessible() function. The only configuration that is required for this target is an API key which I've set up as configurable per search since it sends it to different escalation policies. So I guess just return true?

File: phplib/Target/PagerDuty.php (hunk 2 above)
Comment (kiwiz) [inline phplib/Target/PagerDuty.php]: 👍

**Comment by kiwiz** [comment]
Do you mean the API token? https://v2.developer.pagerduty.com/docs/authentication
I don't see anything in the code code that would set that, just the service key.

**Comment by nevins-b** [comment]
So when you create an event via the API (https://v2.developer.pagerduty.com/docs/trigger-events) you have to pass in a "service_key" which is associated with the service and escalation policy that's defined for your teams in PagerDuty, it basically acts as both authentication and internal routing. There's no additional api keys or tokens which are required to call their API.

**Comment by kiwiz** [comment]
Ah, I see. No need to implement `isAccessible`, then.

**Comment by kiwiz** [comment]
Merged, thanks! :]

**Pull request merged**